### PR TITLE
[8.x] Changing the dump and restore method for a PostgreSQL database

### PR DESCRIPTION
**Pull request by d-stephane**
PR related to issue #34281

### CODE REVIEW

File: src/Illuminate/Database/Console/DumpCommand.php [hunk 1]
```diff
@@ -73,7 +73,7 @@ protected function schemaState(Connection $connection)
      */
     protected function path(Connection $connection)
     {
-        return tap($this->option('path') ?: database_path('schema/'.$connection->getName().'-schema.sql'), function ($path) {
+        return tap($this->option('path') ?: database_path('schema/'.$connection->getName().'-schema.dump'), function ($path) {
```
Comment (taylorotwell) [inline src/Illuminate/Database/Console/DumpCommand.php:76]: Why?

Comment (d-stephane) [reply src/Illuminate/Database/Console/DumpCommand.php:76]: The result obtained with ** pg_dump -Fc ** is not SQL therefore not readable like classic SQL ...

**Comment by taylorotwell** [comment]
I wasn't able to reproduce this issue so don't feel comfortable changing the code at this time. Need more feedback from Pgsql users as to specifically what is happening.

**Comment by d-stephane** [comment]
No problem

**Comment by mfn** [comment]
PgSQL user here.

I see no problem with the PR and in fact would also argue it's more idiomatic to use `-Fc` (performance) and not name it `.sql`, because it simply isn't.

**Comment by taylorotwell** [comment]
@mfn @d-stephane ever since this PR Postgres schema dump has been totally broken to be honest.

First, the migration data is never stored in the dump because this entire method no longer works:

image

It doesn't work because obviously this binary dump doesn't include SELECT statements it can parse out. This causes the subsequent migrations to break because Laravel thinks it always needs to reload the state because the migrations table is empty.

**Comment by taylorotwell** [comment]
I have created a new PR to try to fix this: https://github.com/laravel/framework/pull/35018

**Comment by d-stephane** [comment]
I just tested, it works very well :)

**Pull request merged**